README-gradle.md

```diff
@@ -0,0 +1,191 @@
+## Using iDFlakies on a Gradle project
+### Dependency Requirements
+1. Junit 4/5 required, due to the original iDFlakies' dependency requirement.
+2. Scala (if the project uses Scala) version should be < 2.13, due to the original iDFlakies' dependency requirement and backward incompatibility for some Scala functions.
+3. Gradle (if kotlin is used): [5.0, 6.5.1], due to gradle internal bugs: https://github.com/gradle/gradle/issues/14727. Otherwise, there's no requirement on Gradle version. 
+### Automatically setting up the build.gradle for iDFlakies
+Run the following command to  automatically setup the build.gradle for iDFlakies.
+```
+bash gradle_modify/modify_gradle.sh path_to_gradle_project
+```
+### Manually setting up the build.gradle for iDFlakies
+We can configure the build.gradle file manually instead. We need to follow the rule listed below to successfully set up the project. 
+1. First we need to add the following dependencies and repository to the `buildscript{}` in build.gradle 
+   ```groovy
+   buildscript {
+      repositories {
+         ...
+         mavenCentral()
+      }
+   
+      dependencies {
+         ...
+         classpath group: 'edu.illinois.cs',
+         name: 'idflakies', 
+         /* use iDFlakies from Maven Central */
+         version: '1.1.0'
+         /* use the following version instead if you have built iDFlakies locally and want to use the locally built version*/
+         /* version: '1.2.0-SNAPSHOT'*/
+         /* For instructions on building iDFlakies locally, please see 'Use locally built version of iDFlakies' section below in this document. */
+         
+         classpath group: 'edu.illinois.cs',
+         name: 'testrunner-gradle-plugin', 
+         version: '1.2'
+         ...
+      }
+   }
+   ```
+2. Then, add `apply plugin: 'testrunner'` to the build.gradle file. The location of this code snippet depends on whether build.gradle file has `allprojects{}` and `subprojects{}`. 
+   * If there are both `allprojects{}` and `subprojects{}` methods in gradle.build, then the `apply plugin: "testrunner"` command should be added to the end of  `subprojects{}` and the root (outside of all methods in the build file).
+     * Example: 
+       ```groovy
+       buildscript {
+          repositories {
+             mavenCentral()
+          }
+       
+          dependencies {
+             classpath group: 'edu.illinois.cs',
+             name: 'idflakies', 
+             version: '1.1.0'
+       
+             classpath group: 'edu.illinois.cs',
+             name: 'testrunner-gradle-plugin', 
+             version: '1.2'
+          }
+       }
+       
+       allprojects {
+         repositories {
+         	mavenCentral()
```

Comment:
Please check the indentation of the code samples. ```mavenCentral()``` here seem to be indented more than line 44